ui-components/src/commonMain/kotlin/org/jetbrains/kotlinconf/ui/components/MainHeader.kt

```diff
@@ -0,0 +1,258 @@
+import androidx.compose.animation.togetherWith
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.graphics.SolidColor
+import androidx.compose.ui.unit.dp
+import kotlinconfapp.ui_components.generated.resources.Res
+import kotlinconfapp.ui_components.generated.resources.arrow_left_24
+import kotlinconfapp.ui_components.generated.resources.bookmark_24
+import kotlinconfapp.ui_components.generated.resources.close_24
+import kotlinconfapp.ui_components.generated.resources.search_24
+import org.jetbrains.compose.ui.tooling.preview.Preview
+import org.jetbrains.kotlinconf.ui.theme.KotlinConfTheme
+import org.jetbrains.kotlinconf.ui.theme.PreviewHelper
+
+@Composable
+fun MainHeaderSearchBar(
+    searchValue: String,
+    onSearchValueChange: (String) -> Unit,
+    onClose: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Row(
+        modifier = modifier
+            .height(48.dp)
+            .fillMaxWidth(),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        TopMenuButton(
+            icon = Res.drawable.arrow_left_24,
+            onClick = {
+                onClose()
+                onSearchValueChange("")
+            },
+            contentDescription = "Back",
+        )
+
+        val focusRequester = remember { FocusRequester() }
+        LaunchedEffect(Unit) {
+            focusRequester.requestFocus()
+        }
+
+        Box(
+            modifier = Modifier.weight(1f).fillMaxHeight(),
+            contentAlignment = Alignment.CenterStart,
+        ) {
+            BasicTextField(
+                value = searchValue,
+                onValueChange = { onSearchValueChange(it) },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .focusRequester(focusRequester),
+                singleLine = true,
+                textStyle = KotlinConfTheme.typography.text1
+                    .copy(color = KotlinConfTheme.colors.primaryText),
+                cursorBrush = SolidColor(KotlinConfTheme.colors.primaryText),
+            )
+            androidx.compose.animation.AnimatedVisibility(
```

Comment:
Sadly, that doesn't solve the issue, it would still be ambiguous.

The problem is that there are several functions with the exact same import, `androidx.compose.animation.AnimatedVisibility`:

- [`AnimatedVisibility()`](https://developer.android.com/reference/kotlin/androidx/compose/animation/package-summary#AnimatedVisibility(kotlin.Boolean,androidx.compose.ui.Modifier,androidx.compose.animation.EnterTransition,androidx.compose.animation.ExitTransition,kotlin.String,kotlin.Function1))
- [`RowScope.AnimatedVisibility()`](https://developer.android.com/reference/kotlin/androidx/compose/animation/package-summary#(androidx.compose.foundation.layout.RowScope).AnimatedVisibility(kotlin.Boolean,androidx.compose.ui.Modifier,androidx.compose.animation.EnterTransition,androidx.compose.animation.ExitTransition,kotlin.String,kotlin.Function1))
- [`ColumnScope.AnimatedVisibility()`](https://developer.android.com/reference/kotlin/androidx/compose/animation/package-summary#(androidx.compose.foundation.layout.ColumnScope).AnimatedVisibility(kotlin.Boolean,androidx.compose.ui.Modifier,androidx.compose.animation.EnterTransition,androidx.compose.animation.ExitTransition,kotlin.String,kotlin.Function1))

So whether named or not, if here we call the function with the `AnimatedVisibility()` syntax, it will resolve to the extension on `RowScope` instead of the top level function. The fully qualified name prevents it from using the implicit receiver.

Here's a [StackOverflow question](https://stackoverflow.com/questions/67975569/why-cant-i-use-animatedvisibility-in-a-boxscope) on the same problem, and a [YouTrack issue](https://youtrack.jetbrains.com/issue/KT-48215).